pennylane/ops/qubit/special_unitary.py

```diff
@@ -298,32 +434,240 @@ def compute_matrix(theta, num_wires):
+        inner product (:math:`P_m` is Hermitian, so we skip the adjoint :math:`{}^\dagger`):
+
+        .. math::
+
+            \omega_m = \frac{1}{2^n}\operatorname{tr}\left[P_m \Omega \right]
+
+        The coefficients satisfy :math:`\omega_m^\ast=-\omega_m` because :math:`\Omega` is
+        skew-Hermitian. Therefore they are purely imaginary.
+
+        .. warning::
+
+            An auto-differentiation framework is required by this function.
+            The matrix exponential is not differentiable in Autograd. Therefore this function
+            only supports JAX, Torch and Tensorflow.
+
+        ..seealso:: `~.transforms.insert_paulirot.get_one_parameter_generators`
+
+        """
+        num_wires = self.hyperparameters["num_wires"]
+        basis = pauli_basis_matrices(num_wires)
+        generators = self.get_one_parameter_generators(interface)
+        return qml.math.tensordot(basis, generators, axes=[[1, 2], [2, 1]]) / 2**num_wires
+
+    def decomposition(self):
+        r"""Representation of the operator as a product of other operators.
+
+        .. math:: O = O_1 O_2 \dots O_n
 
         Returns:
             list[Operator]: decomposition of the operator
 
         **Example:**
 
         >>> theta = np.array([0.5, 0.1, -0.3])
-        >>> qml.SpecialUnitary.compute_decomposition(theta, 0, num_wires=1)
+        >>> qml.SpecialUnitary(theta, wires=[0]).decomposition()
         [QubitUnitary(array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
             [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]]), wires=[0])]
         """
-        return [qml.QubitUnitary(qml.SpecialUnitary.compute_matrix(theta, num_wires), wires=wires)]
+        theta = self.data[0]
+        if qml.math.requires_grad(theta):
+            interface = qml.math.get_interface(theta)
+            # Get all Pauli words for the basis of the Lie algebra for this gate
+            words = pauli_basis_strings(self.hyperparameters["num_wires"])
+
+            # Compute the linear map that transforms between the Pauli basis and effective generators
+            # Consider the mathematical derivation for the prefactor 2j
+            omega = qml.math.real(2j * self.get_one_parameter_coeffs(interface))
+
+            # Create zero parameters for each Pauli rotation gate that take over the trace of theta
+            detached_theta = _detach(theta, interface)
+            zeros = theta - detached_theta
+            # Apply the linear map omega to the zeros to create the correct preprocessing Jacobian
+            zeros = qml.math.tensordot(omega, zeros, axes=[[1], [0]])
+
+            # Apply Pauli rotations that yield the Pauli basis derivatives
+            paulirots = [
+                TmpPauliRot(zero, word, wires=self.wires, id="SU(N) byproduct")
+                for zero, word in zip(zeros, words)
+            ]
+            return paulirots + [SpecialUnitary(detached_theta, wires=self.wires)]
```

Comment:
That is something that I don' t really understand, why is the decomposition different when you have diff coeffs. Is it still the decomposition of the SpecialUnitary? 

Is it like separating coefficients and the differentiation (paulirots)?

Reply:
> Is it like separating coefficients and the differentiation (paulirots)?

Exactly! It's basically like saying $\frac{d}{dx} e^{x}|_{x=3}=e^{3} \frac{d}{dx} e^{x}|_{x=0}$, i.e. we keep the "coefficient position" in $e^3$ but use the differentiability at zero. While in this example, there is no gain in doing this separation, for `SpecialUnitary` there is! :)